skada/deep/base.py

```diff
@@ -386,6 +387,45 @@ def predict(self, X, sample_domain=None, **predict_params):
             X["sample_domain"] = sample_domain
         return super().predict(X, **predict_params)
 
+    def predict_proba(self, X, sample_domain=None, **predict_params):
+        """model prediction
+
+        Parameters
+        ----------
+        X : dict, torch tensor, array-like or torch dataset
+            The input data. If a dict, it should contain a key 'X' with the
+            input data and a key 'sample_domain' with the domain of each
+            sample.
+            If X is a dataset, the dataset should return a dict.
+        sample_domain : torch tensor
+            The domain of each sample.
+            Could be None since the sample are not used in predict.
+        """
+        if isinstance(X, dict):
+            if "X" not in X.keys():
```

Comment:
You can easily coverage these lines by duplicating the `test_base` for `predict_proba`